DAS/das/utils/logger.py

```diff
@@ -0,0 +1,275 @@
+from pathlib import Path
+import paho.mqtt.client as mqtt
+import paho.mqtt.publish as publish
+import csv
+import time
+import os
+import asyncio
+import logging
+import re
+
+CsvConfig = {
+    "delimiter": ",",
+    "quotechar": "`",
+    "quoting": csv.QUOTE_ALL,
+    "skipinitialspace": True,
+    "fieldnames": ["time_delta", "mqtt_topic", "message"],
+}
+
+# Set logging to output all info by default (with a space for clarity)
+logging.basicConfig(format="%(levelname)s: %(message)s", level=logging.INFO)
+
+
+class Recorder:
+    """Record incoming MQTT messages on selected topics.
+
+    Parameters
+    ----------
+    csv_folder_path : str
+        Filepath of the folder where the logs are to be stored
+    topics : List(str)
+        A list containing the topic strings that are to be subscribed to
+    broker_address : str
+        The IP address that the MQTT broker lives on
+    verbose : bool
+        Specifies whether the incoming MQTT data and warnings are printed
+
+    Attributes
+    ----------
+    TOPICS : List(str)
+        List of topic names
+    _recording : bool
+        Whether the Recorder object is currently recording or not
+    _VERBOSE : bool
+        Specifies whether the incoming MQTT data and warnings are printed
+    _START_TIME : `time`
+        The current time used to produce time deltas
+    _LOG_FILE : `File`
+        Open log file object that can be written to
+    _LOG_FILE_WRITER : `csv.DictWriter`
+        Csv writter object that is used to write the data to the _LOG_FILE file
+    _CLIENT : `paho.mqtt.client`
+        MQTT client that connects to the broker and recives the messages
+    """
+
+    def __init__(
+        self,
+        csv_folder_path: str,
+        topics: list = ["#"],
+        broker_address: str = "localhost",
+        verbose: bool = False,
+    ) -> None:
+        # The logger object can subscribe to many topics (if none are selected then it will subscribe to all)
+        self.TOPICS = topics
+        self._VERBOSE = verbose
```

Comment:
Instead of having the `if self._VERBOSE:` everywhere, you can actually tell the `logger` library to hide certain levels of logging (so can automatically hide all the info and debug logs).

``` python
if (verbose):
    logging.getLogger().setLevel(logging.ERROR)
```

I think the above might work. pls test.

Reply:
Works a treat!